File: habit-tracker/src/main/java/com/habit/habit_tracker/events/EventPublisher.java
package com.habit.habit_tracker.events;

import java.time.LocalDate;

import org.springframework.context.ApplicationEventPublisher;
import org.springframework.stereotype.Component;

@Component
public class EventPublisher {

    private final ApplicationEventPublisher applicationEventPublisher;

    public EventPublisher(ApplicationEventPublisher applicationEventPublisher) {
        this.applicationEventPublisher = applicationEventPublisher;
    }

    public void publishDHLUpdated(Long habitId, Integer minutesDone, LocalDate date) {
        applicationEventPublisher.publishEvent(new DHLUpdatedEvent(habitId, minutesDone, date));
    }
}

File: habit-tracker/src/main/kotlin/com/habit/habit_tracker/events/EventPublisher.kt
package com.habit.habit_tracker.events

import org.springframework.context.ApplicationEventPublisher
import org.springframework.stereotype.Component
import java.time.LocalDate

@Component
class EventPublisher(
    private val applicationEventPublisher: ApplicationEventPublisher
) {
    fun publishDailyHabitLogUpdated(habitId: Long, minutesDoneChange: Int, date: LocalDate) {
        applicationEventPublisher.publishEvent(DailyHabitLogUpdatedEvent(habitId, minutesDoneChange, date))
    }
}